ACKategories-iOS/Base/FlowCoordinator.swift

```diff
@@ -72,9 +72,11 @@ extension Base {
             let animationGroup = DispatchGroup()
 
             // stop all children
-            childCoordinators.forEach {
-                animationGroup.enter()
-                $0.stop(animated: animated, completion: animationGroup.leave)
+            DispatchQueue.main.async { [weak self] in
```

Comment:
I think we should hold `self` strongly. There is no reference cycle + holding the `self` strongly ensures that it's not deallocated before we stop the child coordinators which is the desired behavior here.

Reply:
Definitely, resolved in 6177daf.

Reply:
I have also extended unit tests to replicate this exact case in 15f17c3. 🙂

Reply:
Great! But to hold `self` strongly, you need to drop `weak` keyword, instead of unwrapping it with `guard`

Reply:
And thanks for writing those tests 👏

Reply:
Oh, okay, sorry for misunderstanding. Modified in b2d2a03. 🙂